Remove 'make install' to avoid confusion in build instructions

jhp-lanl opened the pull request:
<!--Provide a general summary of your changes in the title above, for
example "fix bug in ideal gas EOS.".  Please avoid
non-descriptive titles such as "Addresses issue #8576".-->

## PR Summary

This is a very simple update to the documentation to remove `make install` since it seems to cause confusion.

I've left it in the python build instructions because I assume it's needed there to install the appropriate python bindings.

<!--Please provide at least 1-2 sentences describing the pull request in
detail.  Why is this change required?  What problem does it solve?-->

<!--If it fixes an open issue, please link to the issue here.-->

## PR Checklist

<!-- Note that some of these check boxes may not apply to all pull requests -->

- N/A Adds a test for any bugs fixed. Adds tests for new features.
- N/A Format your changes by using the `make format` command after configuring with `cmake`.
- [x] Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.
- N/A After creating a pull request, note it in the CHANGELOG.md file
- N/A If preparing for a new release, update the version in cmake.

Yurlungur commented:
I'm confused. Why does `make install` not work? It's fine to remove it... but I think typing `make install` does work if you have the right permissions, or if you set 
```
-DCMAKE_INSTALL_PREFIX:PATH=/path/to/target/directory
```
so that you install to some user directory rather than globally.

rbberger commented:
How about adding it like this?
```bash
make
make install    # install into directory defined via CMAKE_INSTALL_PREFIX
```

Yurlungur commented:
:+1: for @rbberger 's suggestion

jhp-lanl commented:
> I'm confused. Why does `make install` not work? It's fine to remove it... but I think typing `make install` does work if you have the right permissions, or if you set
> 
> ```
> -DCMAKE_INSTALL_PREFIX:PATH=/path/to/target/directory
> ```
> 
> so that you install to some user directory rather than globally.

See, _that's_ not documented clearly and many people _don't_ have the right permissions which is why it's confusing. I'll follow Richard's suggestion

jhp-lanl commented:
@rbberger and @Yurlungur I added back in the `make install` command with the appropriate CMake variable in the comment. I also added that it's optional for the basic build procedure since it definitely is optional in most cases.

jhp-lanl commented:
FYI @gopsub although this is merged, feel free to comment on this if there is anything else confusing or unclear about `make install`, and we can add more changes in a future PR.

If there are other confusing aspects of the build process you would like us to document better, feel free to create an issue.
